pkg/bootstrap/config.go

```diff
@@ -449,16 +471,13 @@ func getNodeMetaData(envs []string, plat platform.Environment, nodeIPs []string,
 	meta := &model.BootstrapNodeMetadata{}
 	untypedMeta := map[string]interface{}{}
 
-	extractMetadata(envs, IstioMetaPrefix, func(m map[string]interface{}, key string, val string) {
-		m[key] = val
-	}, untypedMeta)
+	if pc != nil && pc.ProxyMetadata != nil {
```

Comment:
@howardjohn – Are we OK with allowing the mesh config file to set metadata for Pilot to read but not being able to use it to set feature flags for the agent... or would this be even more confusing?

Reply:
I think this is ok, but do we have a way to also set the feature flags for VMs?

It should also probably have a comment around this  and maybe the code that reads proxy config since its not obvious

Reply:
`cluster.env`/`sidecar.env` files are really the only way. 

Inn the future, we probably won't have a cli flag for every customization to meta/feature flags in the `workload` command, so I think the VM docs deserve a blurb about "further customizations to the VM" and that can talk about what each of these files is responsible for. 

I'll add comments to the code as well.

Reply:
We could read the proxy.istio.io/config annotation on workload group though possibly? That would require them to create WG manually rather than via CLI though which is ok?

Reply:
That works but I don't understand what problem it's solving. AFAICT the two issues we have are

1. `istioctl workload entry` sets a lot of metadata in the proxyConfig file. We could just as easily have the output be `cluster.env`. 
1. `istioctl workload entry` generates a proxy config file and it's unclear what parts of it are actually used.

Reply:
> but does force users to rely on istioctl as a translator from proxyConfig

Isn't that the point? User manages a WorkloadGroup (which includes proxy config annotation, like pods), then istioctl handles the implementation details of how we configure the agent?

Would we solve this by just writing proxyMetadata into sidecar.env? That effectively mirrors what k8s does in the injection. Then the VM case is not special

Reply:
To solve the "further customizations" this would work to use as a "base" proxyConfig and any of the generated stuff that the `workload entry` command does would get applied on top of this. 

Anything in defaultConfig.proxyMetadata would also be written to `cluster.env` so that it can be used as feature flags. One question is what takes precedence, ISTIO_META values from proxyConfig or from the environment?

Reply:
If we put `defaultConfig.proxyMetadata`  into `cluster.env` then do we even need this code change here to read directly from proxyMetadata? Then we don't even need to worry about precedence?